server/src/index.ts

```diff
@@ -91,6 +91,15 @@ const createTransport = async (req: express.Request): Promise<Transport> => {
       const value = req.headers[key];
       headers[key] = Array.isArray(value) ? value[value.length - 1] : value;
     }
+    
+    // If the header "x-custom-auth-header" is present, use its value as the custom header name.
+    if (req.headers["x-custom-auth-header"] !== undefined) {
+      const customHeaderName = req.headers["x-custom-auth-header"] as string;
+      if (req.headers[customHeaderName] !== undefined) {
+        const value = req.headers[customHeaderName];
+        headers[customHeaderName] = value as string;
+      }
+    }
```

Comment:
This exact same code appears at lines 132..139. Can we refactor/extract it into a function?

Reply:
Done